File: internal/infrastructure/ssm/ssm.go
package ssm

import (
	"context"
	"fmt"

	"github.com/aws/aws-sdk-go-v2/aws"
	"github.com/aws/aws-sdk-go-v2/service/ssm"
)

type SSMParameterStore struct {
	client *ssm.Client
}

func NewSSMParameterStore(ctx context.Context, cfg aws.Config) (*SSMParameterStore, error) {
	return &SSMParameterStore{
		client: ssm.NewFromConfig(cfg),
	}, nil
}

func (s *SSMParameterStore) Get(ctx context.Context, name string) (string, error) {
	input := &ssm.GetParameterInput{
		Name:           aws.String(name),
		WithDecryption: aws.Bool(true), // auto-decrypt SecureString
	}
	output, err := s.client.GetParameter(ctx, input)
	if err != nil {
		return "", fmt.Errorf("SSM GetParameter failed for %s: %w", name, err)
	}
	if output.Parameter == nil || output.Parameter.Value == nil {
		return "", fmt.Errorf("parameter %s not found or empty", name)
	}
	return *output.Parameter.Value, nil
}

File: tracking/src/aws/ssm.rs
use aws_sdk_ssm::Client;

pub struct SsmParameterStore {
    client: Client,
}

impl SsmParameterStore {
    pub fn new(config: &aws_config::SdkConfig) -> Self {
        Self {
            client: Client::new(config),
        }
    }

    pub async fn get(&self, name: &str) -> Result<String, Box<dyn std::error::Error + Send + Sync>> {
        let output = self
            .client
            .get_parameter()
            .name(name)
            .with_decryption(true)
            .send()
            .await?;

        output
            .parameter
            .and_then(|p| p.value)
            .ok_or_else(|| format!("Parameter {} not found", name).into())
    }

    pub async fn get_optional(&self, name: &str) -> Option<String> {
        self.get(name).await.ok()
    }
}
